modules/scripting/src/main/java/org/mule/module/scripting/transformer/ScriptTransformer.java

```diff
@@ -22,6 +24,20 @@
     private Scriptable script;
 
     @Override
+    public void initialise() throws InitialisationException
+    {
+        super.initialise();
+        LifecycleUtils.initialiseIfNeeded(script, muleContext);
+    }
+
+    @Override
+    public void dispose()
+    {
+        super.dispose();
+        LifecycleUtils.disposeIfNeeded(script);
```

Comment:
What is the reason disposeIfNeeded is invoked sometimes with a logger and sometimes without it?